### PR TITLE
IDEMPIERE-5418 - Add Translation for AD_ChartDatasource

### PR DESCRIPTION
**Pull request by PeterTakacs300**
JIRA ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5418

### CODE REVIEW

**Comment by hengsin** [comment]
Hi @PeterTakacs300 , please document the new feature in wiki.

**Comment by norbertbede** [comment]
@hengsin i can create documentation, however this looks me not new features, simple improvement _trl option.
if you don't agree i will do it.

File: migration/iD10/oracle/202209121558_IDEMPIERE-5418.sql [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+;
+
+-- Sep 12, 2022, 4:00:02 PM CEST
+CREATE TABLE AD_ChartDatasource_Trl (AD_Client_ID NUMBER(10) NOT NULL)
+;
+
+-- Sep 12, 2022, 4:00:37 PM CEST
+INSERT INTO AD_Column (AD_Column_ID,Version,Name,AD_Table_ID,ColumnName,FieldLength,IsKey,IsParent,IsMandatory,IsTranslated,IsIdentifier,SeqNo,IsEncrypted,AD_Reference_ID,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,AD_Element_ID,IsUpdateable,IsSelectionColumn,EntityType,IsSyncDatabase,IsAlwaysUpdateable,IsAutocomplete,IsAllowLogging,AD_Column_UU,IsAllowCopy,SeqNoSelection,IsToolbarButton,IsSecure,IsHtml) VALUES (215306,0,'Chart Datasource',200361,'AD_ChartDatasource_ID',22,'N','Y','Y','N','N',0,'N',30,0,0,'Y',TO_TIMESTAMP('2022-09-12 16:00:36','YYYY-MM-DD HH24:MI:SS'),100,TO_TIMESTAMP('2022-09-12 16:00:36','YYYY-MM-DD HH24:MI:SS'),100,54269,'N','N','D','N','N','N','Y','2d9f86a9-ce57-45b2-932b-4610818f99e7','Y',0,'N','N','N')
```
Comment (CarlosRuiz-globalqss) [inline migration/iD10/oracle/202209121558_IDEMPIERE-5418.sql:28]: The configuration on translation tables is to define the constraints cascade, to be deleted automatically when the parent record is deleted.

File: migration/iD10/oracle/202209121558_IDEMPIERE-5418.sql [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+-- Sep 12, 2022, 4:00:40 PM CEST
+UPDATE AD_Column SET IsUpdateable='N', FKConstraintName='ADChartDatasource_ADChartDatas', FKConstraintType='N',Updated=TO_TIMESTAMP('2022-09-12 16:00:40','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=215306
+;
+
+-- Sep 12, 2022, 4:00:40 PM CEST
+ALTER TABLE AD_ChartDatasource_Trl ADD AD_ChartDatasource_ID NUMBER(10) NOT NULL
+;
+
+-- Sep 12, 2022, 4:00:40 PM CEST
+ALTER TABLE AD_ChartDatasource_Trl ADD CONSTRAINT ADChartDatasource_ADChartDatas FOREIGN KEY (AD_ChartDatasource_ID) REFERENCES ad_chartdatasource(ad_chartdatasource_id) DEFERRABLE INITIALLY DEFERRED
```
Comment (CarlosRuiz-globalqss) [inline migration/iD10/oracle/202209121558_IDEMPIERE-5418.sql:40]: changing the FK to cascade must change this statement too - both in oracle and postgresql

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]

**Comment by CarlosRuiz-globalqss** [comment]
@PeterTakacs300 - I noticed the scripts are doing nothing to translate the current charts.

The attached patch here:
[IDEMPIERE-5418_pr1.txt](https://github.com/idempiere/idempiere/files/9776134/IDEMPIERE-5418_pr1.txt)

Implements the translation on all translatable elements in the "Sales Rep Dashboard"
It additionally fixes the problem with the cascade foreign key requested above.


@hengsin - can you please take a look to the patch too?
It's related to the billboard implementation, I didn't find a way to translate the month names in the "Sales YTD" chart, so I changed the format from "Aug 2022" to "2022-08" - is that acceptable?  Or is there a way to show the names in the language the user is logged in?

**Comment by hengsin** [comment]
hi @CarlosRuiz-globalqss , since I'm not actually using billboard in production, it is more whether that's acceptable to @norbertbede  and perhaps @nmicoud. anyway, I will check whether that's translatable to the login locale.

**Comment by nmicoud** [comment]
hi,
I'm still using the "old" charts so never tried billboard ; I think 2022-08 is fine.
And we can check later if that can be translated according to locale, if a customer starts to complain?

**Comment by norbertbede** [comment]
the scope of this PR was only extend chart source by translation. not fix in billboard. Later we can make a PR for billboard fixes. Billboard  js changes are very experimental, so we don't want to create many issues, but a complex one.

**Comment by CarlosRuiz-globalqss** [comment]
> the scope of this PR was only extend chart source by translation. not fix in billboard.

Yes, that's precisely why I reviewed this, it doesn't make sense to add a translatable table if is not used anywhere, and then I noticed that all the translations in charts were not used (not just on billboard, but I think they are not being used also in the jfreechart version, but I didn't try to fix that one).

> Later we can make a PR for billboard fixes. Billboard js changes are very experimental, so we don't want to create many issues, but a complex one.

I already added the patch for @PeterTakacs300 to integrate.

**Review by CarlosRuiz-globalqss — APPROVED** [review]

**Comment by hengsin** [comment]
hi @CarlosRuiz-globalqss , go ahead to merge this if doc is not necessary.

**Pull request merged**